packages/venia-ui/lib/RootComponents/Category/category.js

```diff
@@ -36,14 +44,60 @@ const Category = props => {
 
     const [runQuery, queryResponse] = useLazyQuery(GET_CATEGORY);
     const { loading, error, data } = queryResponse;
+    const { search } = useLocation();
+
+    // Keep track of the search terms so we can tell when they change.
+    const [previousSearch, setPreviousSearch] = useState(search);
```

Comment:
We know when search terms change because `useLocation` will receive a new value from context, triggering a render cycle for this component. In other words, `useLocation` is the single source of truth for search params.

Ordinarily, if we want to store a previous value for some comparison purpose, [we use a ref for that with `useRef`](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/FilterModal/useFilterModal.js#L23). That way, when the source of truth changes (`useLocation`), a render is triggered and we update the ref as an effect, but since a ref update does not trigger a second render, we avoid unnecessary renders (or worse, an infinite loop).

Do we actually require a second render cycle here, necessitating `useState` over `useRef`?

Reply:
Good point! I copied this from the search page which was pretty much [just added](https://github.com/magento/pwa-studio/pull/2037/files#diff-74df0db583d1513b2f2ae327a4776babR38). I'll fix it in both places.